src/ModelManager.ts

```diff
@@ -125,96 +147,181 @@ export class ModelManager {
+        const metaPropertyModelUrl = PathUtils.internalize(pageModelRoot);
 
-                return data;
-            } else {
-                return this._fetchData(rootModelURL).then((rootModel: Model) => {
-                    this.modelStore.initialize(rootModelPath, rootModel);
+        const aemApiHost = this.modelClient.apiHost;
+        const isRemoteApp = PathUtils.isBrowser() && aemApiHost && (PathUtils.getCurrentURL() !== aemApiHost);
+        const currentPathname = !isRemoteApp ? PathUtils.getCurrentPathname() : '';
+        // For remote apps in edit mode, to fetch path via parent URL
 
-                    // Append the child page if the page model doesn't correspond to the URL of the root model
-                    // and if the model root path doesn't already contain the child model (asynchronous page load)
-                    if (!!currentPathname && !!sanitizedCurrentPathname) {
-                        if (!isPageURLRoot(currentPathname, metaPropertyModelUrl) && !hasChildOfPath(rootModel, currentPathname)) {
-                            return this._fetchData(currentPathname).then((model: any) => {
-                                this.modelStore.insertData(sanitizedCurrentPathname, model);
+        const sanitizedCurrentPathname = ((currentPathname && PathUtils.sanitize(currentPathname)) || '') as string;
 
-                                const data = this.modelStore.getData();
+        // Fetch the app root model
+        // 1. consider the provided page path
+        // 2. consider the meta property value
+        // 3. fallback to the model path contained in the URL for the default SPA
+        const rootModelURL = path || metaPropertyModelUrl || sanitizedCurrentPathname;
+        const rootModelPath = PathUtils.sanitize(rootModelURL) || '';
+
+        this._modelPaths = {
+            currentPathname,
+            sanitizedCurrentPathname,
+            metaPropertyModelUrl,
+            rootModelURL,
+            rootModelPath
+        };
+    }
 
-                                triggerPageModelLoaded(data);
+    /**
+     * Fetch page model from store and trigger cq-pagemodel-loaded event
+     * @returns Root page model
+     */
+    private _fetchPageModelFromStore() {
+        const data = this.modelStore.getData();
 
-                                return data;
-                            });
-                        } else {
-                            const data = this.modelStore.getData();
+        triggerPageModelLoaded(data);
 
-                            triggerPageModelLoaded(data);
+        return data;
+    }
 
-                            return data;
+    /**
+     * Sets initialization promise to fetch model if root path is available
+     * Also, to be returned on synchronous initialization
+     */
+    private _setInitializationPromise(rootModelPath: string) {
+        const {
+            rootModelURL
```

Comment:
Is the purpose to override the value of the field that we have set a few lines prior? Can this operation be optimized into a single localized operation? Or is the purpose to change the state conditionally or on-demand?

https://github.com/adobe/aem-spa-page-model-manager/pull/25/files#diff-e202e8d68c9a3c21edead7a69cd78888R246